templates/repo/diff/box.tmpl

```diff
@@ -57,28 +57,6 @@
 				<div>{{ctx.Locale.Tr "repo.pulls.showing_specified_commit_range" (ShortSha .BeforeCommitID) (ShortSha .AfterCommitID)}} - <a href="{{$.Issue.Link}}/files?style={{if $.IsSplitStyle}}split{{else}}unified{{end}}&whitespace={{$.WhitespaceBehavior}}&show-outdated={{$.ShowOutdatedComments}}">{{ctx.Locale.Tr "repo.pulls.show_all_commits"}}</a></div>
 			</div>
 		{{end}}
-		<script id="diff-data-script" type="module">
-			const diffDataFiles = [{{range $i, $file := .DiffFiles}}
-				{Name:"{{$file.Name}}",NameHash:"{{$file.NameHash}}",Status:{{$file.Status}},IsSubmodule:{{$file.IsSubmodule}},IsViewed:{{$file.IsViewed}}},{{end}}];
-
-			const diffData = {
-				isIncomplete: {{.Diff.IsIncomplete}},
-			};
-
-			// for first time loading, the diffFileInfo is a plain object
-			// after the Vue component is mounted, the diffFileInfo is a reactive object
-			let diffFileInfo = window.config.pageData.diffFileInfo || {
-				files:[],
-				fileTreeIsVisible: false,
-				fileListIsVisible: false,
-				isLoadingNewData: false,
-				selectedItem: '',
-			};
-			diffFileInfo = Object.assign(diffFileInfo, diffData);
-			diffFileInfo.files.push(...diffDataFiles);
-			window.config.pageData.diffFileInfo = diffFileInfo;
-		</script>
-		<div id="diff-file-list"></div>
```

Comment:
With this removal, will it not flash wrong content on page load when diff tree is visible?

Reply:
Ah I may be misunderstanding and this was just some data modification that has no reason to be in templates.

Reply:
Yes, it's not related to "content flash" (and this block was marked as "module", it only executes after DOM ready).

Some old PRs used very tricky methods to make the Vue work, now it's the time to make code overall right.

Reply:
Yeah we should remove all such template js, except those cases that serve the purpose of avoiding flash, but those are better migrated to web components.